ui/src/pages/Dashboard/utils.tsx

```diff
@@ -203,3 +242,127 @@ export const addDescriptionToExpandedViewByOptions = (target: Element) => {
         }
     });
 };
+
+export const createNewQueryForDataVolumeInModal = (
+    selectedInput: string,
+    selectedValue: string,
+    query: string
+) => {
+    const selectedLabel = queryMap[selectedInput];
+    const updatedQuery = query.replace('|', `${selectedLabel} = "${selectedValue}" |`);
+
+    return updatedQuery;
+};
+
+export const createNewQueryForNumberOfEventsInModal = (
+    selectedInput: string,
+    selectedValue: string,
+    query: string
+) => {
+    const selectedLabel = queryMapForEvents[selectedInput];
+    const updatedQuery = query.replace('|', `${selectedLabel} = "${selectedValue}" |`);
+
+    return updatedQuery;
+};
+
+function getLicenseUsageSearchParams(globalConfig: GlobalConfig) {
+    const dashboard = globalConfig?.pages.dashboard as {
+        settings: {
+            custom_license_usage?: {
+                determine_by?: string;
+                search_condition?: string[];
+            };
+        };
+    };
+    const inputNames = globalConfig?.pages.inputs?.services;
+    let determineBy = 's';
+    let licUsgCondition = inputNames?.map((item) => `${item.name}*`).join(',');
+
+    try {
+        const licUsgType = dashboard?.settings?.custom_license_usage?.determine_by || 'Source';
+        const licUsgSearchItems = dashboard?.settings?.custom_license_usage?.search_condition;
+        determineBy = queryMap[licUsgType];
+
+        const combinedItems = [
+            licUsgCondition || [],
+            ...(licUsgSearchItems?.map((el) => `"${el}"`) || []),
+        ];
+
+        licUsgCondition = combinedItems.join(',');
+        return [determineBy, licUsgCondition];
+    } catch (error) {
+        // eslint-disable-next-line no-console
+        console.info(
+            'No custom license usage search condition found. Proceeding with default parameters.'
+        );
+        return null;
+    }
+}
+export async function fetchDropdownValuesFromQuery(
+    globalConfig: GlobalConfig
+): Promise<SearchResponse[]> {
+    const licenseUsageParams = getLicenseUsageSearchParams(globalConfig);
+
+    const timeForDataIngestionTableStart = document
+        ?.querySelector('[data-input-id="data_ingestion_input"] button')
+        ?.getAttribute('data-test-earliest');
+    const timeForDataIngestionTableEnd = document
+        ?.querySelector('[data-input-id="data_ingestion_input"] button')
+        ?.getAttribute('data-test-latest');
```

Comment:
Yes, it will work without checking, but it's good practice to add checks to ensure cleaner code.

Reply:
what I meant was if it is worth to add this `timeForDataIngestionTableEnd`  conditionally to query, as if thats undefined it will be something like 
`latest=undefined`.

so in example something like that 
```
let latestQuery = ''
if(timeForDataIngestionTableEnd){
    const latestQuery = `latest=${timeForDataIngestionTableEnd}`
}

const eventAccountQuery = `index=_internal source=*${globalConfig.meta.name}* action=events_ingested earliest=${timeForDataIngestionTableStart} ${latestQuery} | fieldsummary | fields field values | where field IN ("event_account")`;
```

But yea seams like it is not a problem in the end, so feel free to ignore it.